File: DialogFragmentActivity/app/src/main/java/com/example/deakyu/dialogfragmentactivity/MainActivity.java
package com.example.deakyu.dialogfragmentactivity;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity implements CustomDialogFragment.OnInputListener{

    private Button mOpenDialogBtn;
    private TextView mOutput;

    private String input;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mOpenDialogBtn = findViewById(R.id.open_dialog_btn);
        mOutput = findViewById(R.id.output_message);

        mOpenDialogBtn.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                CustomDialogFragment dialog = new CustomDialogFragment();
                dialog.show(getFragmentManager(), CustomDialogFragment.class.getSimpleName());
            }
        });
    }

    @Override
    public void sendInputToActivity(String input) {
        mOutput.setText(input);
    }
}

File: CoroutineKotlinAndroidDemo/app/src/main/java/com/example/deakyu/coroutinekotlinandroiddemo/MainActivity.kt
package com.example.deakyu.coroutinekotlinandroiddemo

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.TextView
import com.jakewharton.retrofit2.adapter.kotlin.coroutines.experimental.CoroutineCallAdapterFactory
import kotlinx.coroutines.experimental.android.UI
import kotlinx.coroutines.experimental.launch
import kotlinx.coroutines.experimental.withContext
import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

class MainActivity : AppCompatActivity() {

    private lateinit var retrofit: Retrofit

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        retrofit = Retrofit.Builder()
                .baseUrl("https://5b86296505c5890014a90985.mockapi.io/")
                .addConverterFactory(GsonConverterFactory.create())
                .addCallAdapterFactory(CoroutineCallAdapterFactory())
                .build()

        val service = retrofit.create(UserService::class.java)
        launch{
            val response = service.getUsers().await()
            if(response.isSuccessful) {
                val users = response.body()
                withContext(UI) {
                    findViewById<TextView>(R.id.tv_username).text = users?.get(0)?.name
                }
                Log.d("COROUTINE", "${users}")
            } else {
                val error = response.message()
            }

        }
    }
}
